deepchecks/base/dataset.py

```diff
@@ -126,6 +144,168 @@ def __init__(self, df: pd.DataFrame,
+                labels column name. If not provided next name will be used - 'target'
+            **kwargs:
+                additional arguments that will be passed to the main Dataset constructor.
+
+        Returns:
+            Dataset: instance of the Dataset
+
+        Raises:
+            ValueError:
+                if receives zero or more than two numpy arrays;
+                if features (args[0]) is not two dimensional numpy array;
+                if labels (args[1]) is not one dimensional numpy array;
+                if features array or labels array is empty;
+                if features and labels arrays are not of the same size;
+
+        Examples
+        --------
+        >>> features = np.array([[0.25, 0.3, 0.3], [0.14, 0.75, 0.3], [0.23, 0.39, 0.1]])
+        >>> labels = np.array([0.1, 0.1, 0.7])
+        >>> dataset = Dataset.from_numpy(features, labels)
+
+        Creating dataset only from features array.
+
+        >>> dataset = Dataset.from_numpy(features)
+
+        Passing additional arguments to the main Dataset constructor
+
+        >>> dataset = Dataset.from_numpy(
+        ...    features, labels,
+        ...    max_categorical_ratio=0.5
+        ... )
+
+        Specifying features and label columns names.
+
+        >>> dataset = Dataset.from_numpy(
+        ...    features, labels,
+        ...    features=['sensor-1', 'sensor-2', 'sensor-3',],
+        ...    label='labels'
+        ... )
+
+        """
+        if len(args) == 0 or len(args) > 2:
+            raise ValueError(
+                "'from_numpy' constructor expecting to receive two numpy arrays (or at least one)."
+                "First array must contains the features and second the labels."
+            )
+
+        def validate_features_array(
+            features_array: np.ndarray,
+            feature_names: t.Optional[t.Sequence[t.Hashable]] = None
+        ) -> t.Tuple[np.ndarray, t.List[t.Hashable]]:
+            message = (
+                "'from_numpy' constructor expecting features (args[0]) "
+                "to be not empty two dimensional array."
+            )
+            if len(features_array.shape) != 2:
+                raise ValueError(message)
+            if features_array.shape[0] == 0 or features_array.shape[1] == 0:
+                raise ValueError(message)
+
+            if feature_names is not None and len(feature_names) != features_array.shape[1]:
+                raise ValueError(
+                    f'{features_array.shape[1]} features were provided '
+                    f'but only {len(feature_names)} name(s) for them`s.'
+                )
+            elif feature_names is None:
+                feature_names = [f'X{index}'for index in range(1, features_array.shape[1] + 1)]
+
+            return features_array, t.cast(t.List[str], feature_names)
+
+        if len(args) == 1:
+            features_array, features_names = validate_features_array(args[0], features)
```

Comment:
maybe instead of creating inner func, can just move this logic before this `if`

Reply:
updated